server/src/main/java/org/opensearch/action/admin/indices/template/get/GetIndexTemplatesResponse.java

```diff
@@ -57,7 +55,7 @@ public GetIndexTemplatesResponse(StreamInput in) throws IOException {
         int size = in.readVInt();
         indexTemplates = new ArrayList<>();
         for (int i = 0; i < size; i++) {
-            indexTemplates.add(0, IndexTemplateMetadata.readFrom(in));
```

Comment:
Removing the position seems to be changing the order of the templates in response

Reply:
While the list order doesn't matter (since the [order](https://github.com/opensearch-project/OpenSearch/blob/63c75d1b1d188a4bc3917169001aa8711ad702ef/server/src/main/java/org/opensearch/cluster/metadata/IndexTemplateMetadata.java#L71) variable handles the actual ordering), prepending at position 0 was actually flipping the order from the sender.

Reply:
👍